Formula/musikcube.rb

```diff
@@ -0,0 +1,42 @@
+class Musikcube < Formula
+  desc "Terminal-based audio engine, library, player and server"
+  homepage "https://musikcube.com"
+  url "https://github.com/clangen/musikcube/archive/0.99.4.tar.gz"
+  sha256 "dd8e74c8c8460250c0da325941db2df0f7dcca77f2c68a0295ab21e81f8dd8a4"
+  license "BSD-3-Clause"
+  head "https://github.com/clangen/musikcube.git", branch: "master"
+
+  depends_on "ccache" => :build
+  depends_on "cmake" => :build
+  depends_on "asio"
+  depends_on "curl"
+  depends_on "ffmpeg"
+  depends_on "game-music-emu"
+  depends_on "gnutls"
+  depends_on "lame"
+  depends_on "libev"
+  depends_on "libmicrohttpd"
+  depends_on "libogg"
+  depends_on "libopenmpt"
+  depends_on "libvorbis"
+  depends_on "ncurses"
+  depends_on "openssl@1.1"
```

Comment:
Does this use OpenSSL 1.1 specifically or can it use 3?

Reply:
As mentioned above, all transitive dependencies (with the exception of `asio`) depend on 1.1.

Reply:
@Homebrew/core How are we handling things like this today? Thanks!

Reply:
We haven't really run into an issue like this because the audit prevents it from happening.

Is `asio` a runtime dependency? It's a header-only library, so I suspect it is build-only. Making this `:build` might fix the audit failure.

Reply:
Good point, it can almost definitely be defined as a build-time dependency. Let's give that a go... https://github.com/Homebrew/homebrew-core/pull/118938/commits/4c23788433bee799e03c526fd874ee8c96ee412b

Reply:
OK, I can't tell if making it a build-time dependency has fixed anything, because I'm seeing the following failure in CI:

```
Full audit musikcube --online --new-formula output
  musikcube:
    * Dependency 'curl' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
    * Dependency 'ncurses' is provided by macOS; please replace 'depends_on' with 'uses_from_macos'.
```

However, as previously mentioned, the version of `ncurses` provided by macOS has some really bad bugs that prevent the app from functioning properly in the popular `kitty` terminal emulator, so I'd very much prefer to use the `ncurses` provided version.

Additionally, I'd like to use the most recent version of `curl` simply because it will always have the most recent security patches.

Is it possible to override this particular test failure so I can see if the `openssl` dependency conflict has been solved by recent commits?

Reply:
Add your formula to https://github.com/Homebrew/homebrew-core/blob/master/audit_exceptions/provided_by_macos_depends_on_allowlist.json in this PR.

Reply:
> I'm fine with using the ncurses formula if using the system one leads to bugs.

Cool, is there something I need to do on my end to move past this failure? 

> Apple ships security patches for system `curl`.

Got it. I've updated `curl` to use `uses_from_macos`

> We can still merge this even if there is a brew audit failure for dependencies.

Nice. I temporarily migrated to `uses_from_macos` to `ncurses` to make sure there are no additional build failures, but have reverted back to `depends_on`. Builds are in flight right now, but I'm expecting them to fail again with the same error.

Is there something additional that needs to be done to ensure builds progress pass this audit error so CI is happy and binaries are generated?